### DISCUSSION TITLE
How to utilize quiltflower as decompiler?

### DISCUSSION THREAD

**Discussion by jay-babu** [mfussenegger/nvim-jdtls#411] [Q&A]
In my config, I currently have `fernflower` as the decompiler. I was wondering if anyone has set up quiltflower successfully? https://github.com/QuiltMC/quiltflower. It seems to be better than `fernflower`. I tried searching github and google, but unable to find any example configuration.

Completely wild guess, maybe this is all I need?
```
contentProvider = {
	preferred = "quiltflower",
},
```

and also add the path to it in my `bundles`.

**Comment by mfussenegger** [comment]
I doubt that will work as there needs to be something that implements eclipse's decompiler interface and a bundle definition to ensure eclipse finds it and knows what to do with it.

For fernflower that's https://github.com/dgileadi/dg.jdt.ls.decompiler/blob/master/dg.jdt.ls.decompiler.common/src/dg/jdt/ls/decompiler/common/CachingDecompiler.java (which is the source of https://github.com/dgileadi/vscode-java-decompiler)

There is an issue about supporting quiltflower in https://github.com/dgileadi/vscode-java-decompiler/issues/16

**Reply by jay-babu** [reply]
Thank you for the explanation! I tried looking through the code. unsure how to go about implementing it myself, so will have to wait
